From 9ca926bc49926832bbda51618f4f50eab7beef0a Mon Sep 17 00:00:00 2001
From: Tom Englund <tomenglund26@gmail.com>
Date: Tue, 24 May 2022 04:53:11 +0200
Subject: [PATCH] fix: ensure we dont endlessly loop in groups

adding the newly created group window as a child to all members and all
its members a child for the window creates cyclic loops in
stacking_order and find_modal, the comments say they should be added as
such according to the ICCCM but all the ICCCM says about this is, It is
up to the window manager to determine the policy for treating the
windows in a group.
---
 win/x11/transient.h | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/win/x11/transient.h b/win/x11/transient.h
index fdcbb4682..c7fea8548 100644
--- a/win/x11/transient.h
+++ b/win/x11/transient.h
@@ -177,7 +177,6 @@ void update_group(Win* win, bool add)
             win->in_group->addMember(win);
         }
         auto const win_is_group_tr = win->groupTransient();
-        auto const win_is_normal_tr = !win_is_group_tr && win->transient()->lead();
 
         // This added window must be set as transient child for all windows that have no direct
         // or indirect transient relation with it (that way we ensure there are no cycles).
@@ -194,14 +193,8 @@ void update_group(Win* win, bool add)
                 // have no cycle.
                 if (!member_is_normal_tr && !member->transient()->is_follower_of(win)) {
                     member->transient()->add_child(win);
-                    continue;
                 }
             }
-
-            if (member_is_group_tr && !win_is_normal_tr
-                && !win->transient()->is_follower_of(member)) {
-                win->transient()->add_child(member);
-            }
         }
     } else {
         win->in_group->ref();
-- 
GitLab

